src/users/usersActions.js

```diff
@@ -33,17 +33,19 @@ export const reportPresence = (hasFocus: boolean = true, newUserInput: boolean =
   });
 };
 
-export const sendTypingEvent = (narrow: Narrow) => async (
+export const sendTypingEvent = (narrow: Narrow, opertation: TypingOperation) => async (
   dispatch: Dispatch,
   getState: GetState,
 ) => {
   if (!isPrivateOrGroupNarrow(narrow)) {
     return;
   }
 
-  if (differenceInSeconds(new Date(), lastTypingStart) > 15) {
+  const isStopEvent: boolean = opertation === 'stop';
+
+  if (differenceInSeconds(new Date(), lastTypingStart) > 15 || isStopEvent) {
     const auth = getAuth(getState());
-    api.typing(auth, narrow[0].operand, 'start');
-    lastTypingStart = new Date();
+    api.typing(auth, narrow[0].operand, opertation);
+    lastTypingStart = isStopEvent ? 0 : new Date();
```

Comment:
Hmm. Seeing the ways this control flow comes out differently for 'stop' vs. 'start', I think it'd make most sense to make this two different functions rather than one function with an argument.

Say, `sendTypingStart` and `sendTypingStop`.